typing/typecore.ml

```diff
@@ -1044,7 +1045,11 @@ and type_pat_aux ~constrs ~labels ~no_existentials ~mode ~explode ~env
         pat_env = !env }
       in
       if explode > 0 then
-        let (sp, constrs, labels) = Parmatch.ppat_of_type !env expected_ty in
+        let (sp, constrs, labels) =
+          try
+            Parmatch.ppat_of_type !env expected_ty
+          with Parmatch.Empty -> raise (Error (loc, !env, Empty_pattern))
```

Comment:
@Drup Seems I misunderstood your review. Exception `raise Error...` sometimes is used to backtrack in this function. So I have to add a new variant `Empty_pattern` in `error` to use `raise Err...`

Reply:
I see, that's a bit inconvenient, but I don't have a better proposition, so I suppose it's fine.